sklearn/ensemble/tests/test_forest.py

```diff
@@ -450,7 +450,8 @@ def check_parallel(name, X, y):
     y1 = forest.predict(X)
     forest.set_params(n_jobs=2)
     y2 = forest.predict(X)
-    assert_array_almost_equal(y1, y2, 3)
+    # decimals=3 failed on some platforms due to parallel summation imprecision
+    assert_array_almost_equal(y1, y2, 2)
```

Comment:
This test is used both for classification and regression. The docstring should probably be updated to reflect that otherwise this is confusing.

Reply:
It's unlikely that summation rounding errors could be that important. 

I think this test failure might reveal a real problem: concurrent array summation might not be thread-safe. I think we need atomic increment semantics but they probably do not exist at the numpy API level.

Reply:
So you don't think this is an issue of summation order, but of the summations sharing memory dirtily??

Reply:
Ah, right, I see. I hadn't noticed (or had forgotten) that we are accumulating in parallel. That's no good...

Right. Needs a different fix.